server/block/leaves.go

```diff
@@ -75,8 +75,8 @@ func (l Leaves) FlammabilityInfo() FlammabilityInfo {
 func (l Leaves) BreakInfo() BreakInfo {
 	return newBreakInfo(0.2, alwaysHarvestable, func(t tool.Tool) bool {
 		return t.ToolType() == tool.TypeShears || t.ToolType() == tool.TypeHoe
-	}, func(t tool.Tool) []item.Stack {
-		if t.ToolType() == tool.TypeShears { // TODO: Silk Touch
+	}, func(stack item.Stack) []item.Stack {
+		if _, ok := stack.Item().(item.Shears); ok || hasSilkTouch(stack) {
```

Comment:
I'm not sure if this change from `tool.Tool` to `item.Stack` is a good one. This change right here makes it impossible to have functional custom shears. Is there some way we can check for silk touch on tools? Maybe we can even pass an additional bool to this drop function which indicates silk touch vs no silk touch?